DataFormats/Detectors/TOF/include/DataFormatsTOF/Cluster.h

```diff
@@ -40,23 +44,26 @@ class Cluster : public o2::BaseCluster<float>
 
   Cluster() = default;
 
-  Cluster(std::int16_t sensid, float x, float y, float z, float sy2, float sz2, float syz, float timeRaw, float time, float tot, int L0L1latency, int deltaBC);
+  Cluster(std::int16_t sensid, float x, float y, float z, float sy2, float sz2, float syz, double timeRaw, double time, float tot, int L0L1latency, int deltaBC);
 
   ~Cluster() = default;
 
-  float getTimeRaw() const { return mTimeRaw; }             // Cluster ToF getter
-  void setTimeRaw(float timeRaw) { mTimeRaw = timeRaw; }    // Cluster ToF setter
-  float getTime() const { return mTime; }                   // Cluster ToF getter
-  void setTime(float time) { mTime = time; }                // Cluster ToF setter
+  void SetBaseData(std::int16_t sensid, float x, float y, float z, float sy2, float sz2, float syz);
+
+  double getTimeRaw() const { return mTimeRaw; }             // Cluster ToF getter
+  void setTimeRaw(double timeRaw) { mTimeRaw = timeRaw; }    // Cluster ToF setter
+  double getTime() const { return mTime; }                   // Cluster ToF getter
+  void setTime(double time) { mTime = time; }                // Cluster ToF setter
   float getTot() const { return mTot; }                     // Cluster Charge getter
   void setTot(int tot) { mTot = tot; }                      // Cluster ToT setter
   int getL0L1Latency() const { return mL0L1Latency; };      // L0L1 latency
   void setL0L1Latency(int value) { mL0L1Latency = value; }; // L0-L1 latency
   int getDeltaBC() const { return mDeltaBC; };              // deltaBC
   void setDeltaBC(int value) { mDeltaBC = value; };         // deltaBC
   //float  getZ()   const   {return mZ;}   // Cluster Z - already in the definition of the cluster
-  float getR() const { return mR; }     // Cluster Radius
-  float getPhi() const { return mPhi; } // Cluster Phi
+  float getR() const { return TMath::Sqrt(getX()*getX() + getY()*getY() + getZ()*getZ()); }     // Cluster Radius
+  float getPhi() const { return TMath::ATan2(getY(), getX()); } // Cluster Phi
+  int getSector() const { return int((TMath::ATan2(-getY(), -getX())+TMath::Pi())*TMath::RadToDeg()*0.05);} // Cluster Sector
```

Comment:
is this function used often? If yes, it could potentially make sense to make the sector a property of the cluster instead of having to do this expensive operation over and over again.

Reply:
Changed, now we calculate sector, phi, r only once.